### PR TITLE
add support for Accept:application/json to /api/v2/metrics

### PR DESCRIPTION
**Pull request by ryanpetrello**
see: https://github.com/ansible/awx/issues/4144

### CODE REVIEW

**Comment by ryanpetrello** [comment]
```
~/dev/awx awx metrics -f yaml | head
awx_custom_virtualenvs_total:
  labels: {}
  value: 0.0
awx_hosts_total:
  labels:
    type: active
  value: 1.0
awx_instance_capacity:
  labels:
    hostname: awx
```

**Comment by wenottingham** [comment]
can probably axe the `ignore_json_errors` bit out of api/pages/page.py

**Comment by kdelee** [comment]
whatchoo doin about this https://github.com/ansible/awx/blob/99357acf5d8e8cba8f4c5e2fcab40ed1455b7f1b/awxkit/awxkit/api/pages/metrics.py#L1-L19

File: awxkit/awxkit/api/pages/metrics.py [hunk 1]
```diff
@@ -8,14 +6,8 @@
...
-        self.page_identity(request, ignore_json_errors=True)
-        parsed_metrics = text_string_to_metric_families(request.text)
-        data = {}
-        for family in parsed_metrics:
-            for sample in family.samples:
-                data[sample[0]] = {"labels": sample[1], "value": sample[2]}
-        request.json = lambda: data
+        request = self.connection.get(self.endpoint, query_parameters,
+                                      headers={'Accept': 'application/json'})
         return self.page_identity(request)
```
Comment (kdelee) [inline awxkit/awxkit/api/pages/metrics.py:11]: oh wait never mind I'm blind, I see now

**Review by kdelee — APPROVED** [review]
:shipit:

File: awxkit/awxkit/api/pages/page.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ def page_identity(self, response, request_json=None, ignore_json_errors=False):
             data = response.json()
         except ValueError as e:  # If there was no json to parse
             data = dict()
-            if (response.text and not ignore_json_errors) or response.status_code not in (200, 202, 204):
```
Comment (ryanpetrello) [inline awxkit/awxkit/api/pages/page.py:194]: @wenottingham originally added this to accommodate the old endpoint (which only returned `text/plain`).

**Review by rooftopcellist — APPROVED** [review]
wonderful!

**Pull request merged**